Generic/MultipleStatementAlignment: fix bug/list syntax before assignment operator

jrfnl opened the pull request:
This simplifies the fixes which were previously committed to fix #1870 and #1848, which were released in `3.2.3`.

Both those fixes try to prevent the sniff getting confused over (potentially) multi-line value assignments, such as closures and arrays.

The simplification removes the previously added code in favour of skipping to the end of a statement as soon as an assignment operator has been found.

This might also make the sniff marginally faster as the tokens between the assignment operator and the end of the statement will no longer be examined.

N.B.: The reason for going one back from the end of the assignment (line 241), is that otherwise the code addressing the issue covered by the unit test on line 220 / added in commit 579d39cac3163bdf05ce1b50a7e470acd452735b would not be triggered, leading to incorrect results.

Fixes #1922

Side-note: I think it would be a good idea to add unit tests for the `File::findEndOfStatement()` method.

iltar commented:
Would it be an idea to add this fixture as well, just to show that if there is _no_ blank line, it should still be aligned?
```php
public function buildForm(FormBuilderInterface $builder, array $options)
{
    $transformer                                    = new ContractTransformer($options['contracts']);
    $types                                          = ['support.contact.question' => ContactData::QUESTION];
    [$important, $questions, $incidents, $requests] = $this->createContractBuckets($options['contracts']);
}
```

jrfnl commented:
> Would it be an idea to add this fixture as well, just to show that if there is no blank line, it should still be aligned?

I've added the additional unit test as a safe-guard.

Just so you know: that test case would not be affected by the fix I've now implemented anyway as the whole "check for arrays" bit has been removed, so this is more a safeguard for the future than that it actually tests something now.

iltar commented:
Fair enough! :+1:
